lib/open_api_spex/schema.ex

```diff
@@ -325,8 +325,10 @@ defmodule OpenApiSpex.Schema do
   included in the `allOf` list.
   """
   def properties(schema = %Schema{type: :object, properties: properties = %{}}) do
-    for({name, property} <- properties, do: {name, default(property)}) ++
-      properties(%{schema | properties: nil})
+    properties
+    |> Enum.map(fn {name, property} -> {name, default(property)} end)
+    |> Enum.concat(properties(%{schema | properties: nil}))
+    |> Enum.uniq_by(fn {name, _property} -> name end)
```

Comment:
Would it be more correct to merge the property schemas here?

Eg if one schema is `%{type: :integer, minimum: 1}` and the other is `%{type: :integer, maximum: 5}` then the result should be `%{type: :integer, minimum: 1, maximum: 5}`.

Reply:
Hello @mbuhot 
I'm not sure if that is needed here. This function returns only list of tuple - where first element is the name of property and second is the default value. As far as I did my `grep` right, this function is used in `OpenApiSpex.DeprecatedCast` so we shouldn't care about this as is deprecated and in macro `OpenApiSpex.schema/2` where is used as argument for `defstruct` so current form of list of tuples is enough there.
But I'm new to this library and source code of it so if I'm wrong, please let me know :)

Reply:
BTW I can add `@spec` here, so it will be clear what this function returns.